signer/src/stacks_api.rs

```diff
@@ -0,0 +1,265 @@
+fn url_deserializer<'de, D>(deserializer: D) -> Result<url::Url, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    String::deserialize(deserializer)?
+        .parse()
+        .map_err(serde::de::Error::custom)
+}
+
+/// A struct for the entries in the signers Config.toml (which is currently
+/// located in src/config/default.toml)
+#[derive(Debug, serde::Deserialize)]
+pub struct StacksSettings {
+    api: StacksApiSettings,
+    node: StacksNodeSettings,
+}
+
+/// Whatever
+#[derive(Debug, serde::Deserialize)]
+pub struct StacksApiSettings {
+    /// TODO(225): We'll want to support specifying multiple Stacks API
+    /// endpoints.
+    #[serde(deserialize_with = "url_deserializer")]
+    endpoint: url::Url,
+}
+
+/// Settings associated with the stacks node that this signer uses for information
+#[derive(Debug, serde::Deserialize)]
+pub struct StacksNodeSettings {
+    /// TODO(225): We'll want to support specifying multiple Stacks Nodes
+    /// endpoints.
+    #[serde(deserialize_with = "url_deserializer")]
+    endpoint: url::Url,
+}
+
+impl StacksSettings {
+    /// Create a new StacksSettings object by reading the relevant entries
+    /// in the signer's config.toml. The values there can be overridden by
+    /// environment variables.
+    ///
+    /// # Notes
+    ///
+    /// The relevant environment variables and the config entries that are
+    /// overridden are:
+    ///
+    /// * SIGNER_STACKS_API_ENDPOINT <-> stacks.api.endpoint
+    /// * SIGNER_STACKS_NODE_ENDPOINT <-> stacks.node.endpoint
+    ///
+    /// Each of these overrides an entry in the signer's `config.toml`
+    pub fn new_from_config() -> Result<Self, Error> {
+        let source = File::with_name("./src/config/default");
+        let env = Environment::with_prefix("SIGNER")
+            .prefix_separator("_")
+            .separator("_");
+
+        let conf = Config::builder()
+            .add_source(source)
+            .add_source(env)
+            .build()
+            .map_err(Error::SignerConfig)?;
+
+        conf.get::<StacksSettings>("stacks")
+            .map_err(Error::StacksApiConfig)
+    }
+}
+
+/// A client for interacting with Stacks nodes and the Stacks API
+pub struct StacksClient {
+    /// The base URL (with the port) that will be used when making requests
+    /// for to the Stacks API.
+    pub api_endpoint: url::Url,
+    /// The base URL (with the port) that will be used when making requests
+    /// for to a Stacks node.
+    pub node_endpoint: url::Url,
+    /// The client used to make the request.
+    pub client: reqwest::Client,
+}
+
+impl StacksClient {
+    /// Create a new instance of the Stacks client using the entries in the
+    /// signer's `config.toml` and environment variables.
+    ///
+    /// See [`StacksSettings::new_from_config`] for more on overriding
+    /// the entries in `config.toml` using environment variables.
+    pub fn new_from_config() -> Result<Self, Error> {
```

Comment:
Nit: I'd like to take the config as a parameter here, making it easier to control the configuration in tests.

```suggestion
    pub fn new(config: StacksSettings) -> Result<Self, Error> {
```

Reply:
Yeah, that makes sense. Will do.